src/main/java/org/aksw/gerbil/io/nif/AnnotationWriter.java

```diff
@@ -129,6 +129,8 @@ public void addRelation(Model nifModel, Resource documentAsResource, String docu
         nifModel.add(relationAsResource, RDF.subject, ResourceFactory.createResource(relation.getSubject().getUris().iterator().next()));
         nifModel.add(relationAsResource, RDF.predicate, ResourceFactory.createResource(relation.getPredicate().getUris().iterator().next()));
         nifModel.add(relationAsResource, RDF.object, ResourceFactory.createResource(relation.getObject().getUris().iterator().next()));
+        nifModel.add(relationAsResource, OA.hasTarget,  OA.SpecificResource);
+        nifModel.add(relationAsResource, OA.hasSource, "");
```

Comment:
These two lines are not creating the triples that we would like to have. Please remember that the relation should have the following structure:
```turtle
[]
  a  rdf:Statement , oa:Annotation ;
  rdf:subject  dbr:Zayn_Malik ;
  rdf:predicate  dbo:formerBandMember ;
  rdf:object  dbr:One_Direction ;
  oa:hasTarget [
    a oa:SpecificResource;
    oa:hasSource <http://www.eonline.com/de/news/639463/zayn-malik-leaves-one-direction-read-his-official-statement-and-get-details-on-the-boy-band-s-future#char=0,32>  ] .
```
Your code currently creates something like
```turtle
[]
  a  rdf:Statement , oa:Annotation ;
  rdf:subject  dbr:Zayn_Malik ;
  rdf:predicate  dbo:formerBandMember ;
  rdf:object  dbr:One_Direction ;
  oa:hasTarget oa:SpecificResource;
  oa:hasSource "" .
```
You have missed to create the blank node that is connecting the relation with the source document (and the URI of the source document is not used correctly).